src/JsonRpcRequest.test.ts

```diff
@@ -0,0 +1,59 @@
+import { is } from 'superstruct';
+
+import { JsonRpcRequestStruct } from './JsonRpcRequest';
+
+describe('JsonRpcRequestStruct', () => {
+  it('should be a valid JsonRpcRequest with a numerical ID', () => {
+    const request = {
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'my_method',
+      params: [1, 2, 3],
+    };
+
+    expect(is(request, JsonRpcRequestStruct)).toBe(true);
+  });
+
+  it('should be a valid JsonRpcRequest with a string ID', () => {
+    const request = {
+      jsonrpc: '2.0',
+      id: 'request_id',
+      method: 'my_method',
+      params: [1, 2, 3],
+    };
+
+    expect(is(request, JsonRpcRequestStruct)).toBe(true);
+  });
+
+  it('should be a valid JsonRpcRequest with a null ID', () => {
+    const request = {
+      jsonrpc: '2.0',
+      id: null,
+      method: 'my_method',
```

Comment:
should we check for undefined to return false?

Reply:
You are right, lemme add this test. (We might wanna add this test to the `keyring-api` too)

Reply:
There you go: https://github.com/MetaMask/chain-api/pull/1/commits/de199d63ae4cf4d836acc1a575a4d341dbd0bc37